src/test/ui/parser/issue-32214.stderr

```diff
@@ -1,8 +1,8 @@
 error: type parameters must be declared prior to associated type bindings
-  --> $DIR/issue-32214.rs:3:37
+  --> $DIR/issue-32214.rs:5:34
    |
-LL | pub fn test<W, I: Iterator<Item=(), W> >() {}
-   |                                     ^
+LL | pub fn test<W, I: Trait<Item=(), W> >() {}
+   |                                  ^
```

Comment:
This should suggest the appropriate place for the type parameter, like we do for lifetimes.